client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -330,6 +337,21 @@ private void fetchUpdatesIfNeeded() {
             compactIfNeeded();
         }
     }
+
+    void updateConfigIfNeeded() {
+        sync.fetchUpdates();
+        ReaderGroupState state = sync.getState();
+        if (!updateConfigTimer.hasRemaining() && state.isUpdatingConfig()) {
+            log.debug("Update the group config");
+            ReaderGroupConfig controllerConfig = getThrowingException(controller.getReaderGroupConfig(sync.getSynchronizerScopeName(),
+                    sync.getSynchronizerStreamName().replace(READER_GROUP_STREAM_PREFIX, "")));
```

Comment:
This precludes cross scope reading, which is an important feature. 
Don't make assumptions about the scope and stream name here. Explicitly obtain them from the state. (Or add them as member variables to this class)

Reply:
The scope name and readergroup name are required to make the controller call. Instead of obtaining it from the sync these are now part of the `ReaderGroupStateManager`. This means we wouldn't need those `StateSynchronizer` API's too.